src/test/java/net/sf/jabref/exporter/MSBibExportFormatTestFiles.java

```diff
@@ -0,0 +1,104 @@
+import net.sf.jabref.model.entry.BibEntry;
+
+import com.google.common.base.Charsets;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+import org.w3c.dom.Document;
+import org.xml.sax.SAXException;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Parameterized.class)
+public class MSBibExportFormatTestFiles {
+
+    public static BibDatabaseContext databaseContext;
+    public static Charset charset;
+    public List<BibEntry> entries;
+    public File tempFile;
+    public static MSBibExportFormat msBibExportFormat;
+    public BibtexImporter testImporter = new BibtexImporter();
+    public static final String PATH_TO_FILE = "src/test/resources/net/sf/jabref/exporter/";
+    public String xmlFileName;
+
+    @Parameter
+    public String filename;
+
+    @Rule
+    public TemporaryFolder testFolder = new TemporaryFolder();
+
+
+    @Parameters(name = "{0}")
+    public static Collection<String> fileNames() throws IOException {
+        List<String> files = new ArrayList<>();
+        try (Stream<Path> stream = Files.list(Paths.get(PATH_TO_FILE))) {
+            files = stream.map(n -> n.getFileName().toString()).filter(n -> n.endsWith(".bib"))
+                    .filter(n -> n.startsWith("MsBibExportFormat")).collect(Collectors.toList());
+        }
+        return files;
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        Globals.prefs = JabRefPreferences.getInstance();
+        databaseContext = new BibDatabaseContext(new BibDatabase(), new MetaData());
+        charset = Charsets.UTF_8;
+        msBibExportFormat = new MSBibExportFormat();
+    }
+
+    @Test
+    public final void testPerformExport() throws IOException {
+        xmlFileName = filename.replace(".bib", ".xml");
+        tempFile = testFolder.newFile();
+        String tempFilename = tempFile.getCanonicalPath();
+        try (InputStream bibIn = MSBibExportFormat.class.getResourceAsStream(filename)) {
+            entries = testImporter.importEntries(bibIn, new OutputPrinterToNull());
+        }
+        assertNotNull(entries);
+        msBibExportFormat.performExport(databaseContext, tempFilename, charset, entries);
+        try (FileInputStream stream = new FileInputStream(tempFile)) {
+            DocumentBuilder documentBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+            Document exportedDocument = documentBuilder.parse(stream, tempFilename);
+            exportedDocument.normalizeDocument();
+            Document expectedDocument = documentBuilder.parse(new File(PATH_TO_FILE + xmlFileName));
+            expectedDocument.normalizeDocument();
+            assertTrue(expectedDocument.isEqualNode(exportedDocument));
```

Comment:
I think our importer and exporter are not good enough to handle such roundtrips. However, comparing the files just as plain strings should work without any problems and should reduce the complexity of the test.